process/transactions/sc_processor_test.go

```diff
@@ -0,0 +1,139 @@
+package transactions_test
+
+import (
+	"github.com/ElrondNetwork/covalent-indexer-go"
+	"github.com/ElrondNetwork/covalent-indexer-go/mock"
+	"github.com/ElrondNetwork/covalent-indexer-go/process/transactions"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestNewSCProcessor(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		args        func() core.PubkeyConverter
+		expectedErr error
+	}{
+		{
+			args: func() core.PubkeyConverter {
+				return nil
+			},
+			expectedErr: covalent.ErrNilPubKeyConverter,
+		},
+		{
+			args: func() core.PubkeyConverter {
+				return &mock.PubKeyConverterStub{}
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, currTest := range tests {
+		_, err := transactions.NewSCProcessor(currTest.args())
+		require.Equal(t, currTest.expectedErr, err)
+	}
+}
+
+func TestScProcessor_ProcessSCs(t *testing.T) {
+	scp, _ := transactions.NewSCProcessor(&mock.PubKeyConverterStub{})
+
+	tx1 := &smartContractResult.SmartContractResult{
+		Nonce:          1,
+		Value:          big.NewInt(2),
+		RcvAddr:        []byte("rcv1"),
+		SndAddr:        []byte("snd1"),
+		RelayerAddr:    []byte("rly1"),
+		RelayedValue:   big.NewInt(3),
+		Code:           []byte("code1"),
+		Data:           []byte("data1"),
+		PrevTxHash:     []byte("prevHash1"),
+		OriginalTxHash: []byte("origHash1"),
+		GasLimit:       4,
+		GasPrice:       5,
+		CallType:       6,
+		CodeMetadata:   []byte("codeMeta1"),
+		ReturnMessage:  []byte("retMsg1"),
+		OriginalSender: []byte("origSnd1"),
+	}
+	tx2 := &smartContractResult.SmartContractResult{
+		Nonce:          7,
+		Value:          big.NewInt(8),
+		RcvAddr:        []byte("rcv2"),
+		SndAddr:        []byte("snd2"),
+		RelayerAddr:    []byte("rly2"),
+		RelayedValue:   big.NewInt(9),
+		Code:           []byte("code2"),
+		Data:           []byte("data2"),
+		PrevTxHash:     []byte("prevHash2"),
+		OriginalTxHash: []byte("origHash2"),
+		GasLimit:       10,
+		GasPrice:       11,
+		CallType:       12,
+		CodeMetadata:   []byte("codeMeta2"),
+		ReturnMessage:  []byte("retMsg2"),
+		OriginalSender: []byte("origSnd2"),
+	}
+	tx3 := &transaction.Transaction{
+		Nonce:       9,
+		Value:       big.NewInt(10),
+		RcvAddr:     []byte("rcv3"),
+		SndAddr:     []byte("snd3"),
+		GasLimit:    11,
+		GasPrice:    12,
+		Signature:   []byte("sig3"),
+		SndUserName: nil,
+		RcvUserName: nil,
+	}
+
+	txPool := map[string]data.TransactionHandler{
+		"hash1": tx1,
+		"hash2": tx2,
+		"hash3": tx3,
+	}
+
+	ret := scp.ProcessSCs(txPool, 123)
+
+	require.Len(t, ret, 2)
+
+	require.Equal(t, ret[0].Hash, []byte("hash1"))
```

Comment:
instead of checking every field from resulted SCR you can create an expectedSmartContractResult with all the fields that are expected